File: 이재진/바탕화면정리.py
def solution(wallpaper):
    ls = []
    for i in range(len(wallpaper)):
        for j in range(len(wallpaper[i])):
            if wallpaper[i][j] == "#":
                ls.append((i,j))
    left_i = 100
    left_j = 100
    right_i = 0
    right_j = 0
    for i in ls:
        if left_i > i[0]:
            left_i = i[0]
        if right_i < i[0]:
            right_i = i[0]
        if left_j > i[1]:
            left_j = i[1]
        if right_j < i[1]:
            right_j = i[1]
    answer = [left_i, left_j, right_i+1, right_j+1]
    
    print(ls)
    return answer

File: 정종길/프로그래머스/기타/바탕화면 정리.java
class Solution {
    public int[] solution(String[] wallpaper) {
        int si = -1, sj = -1, ei = -1, ej = -1;
        int row = wallpaper.length;
        int col = wallpaper[0].length();

        for(int i = 0; i < row; i++){
            for(int j = 0; j < col; j++){
                if(si == -1 && wallpaper[i].charAt(j) == '#'){
                    si = i;
                    break;
                }
            }
        }

        for(int j = 0; j < col; j++){
            for(int i = 0; i < row; i++){
                if(sj == -1 && wallpaper[i].charAt(j) == '#'){
                    sj = j;
                    break;
                }
            }
        }


        for(int i = row - 1; 0 <= i ; i--){
            for(int j = col - 1 ; 0 <= j; j--){
                if(ei == -1 && wallpaper[i].charAt(j) == '#'){
                    ei = i + 1;
                    break;
                }
            }
        }


        for(int j = col - 1 ; 0 <= j; j--){
            for(int i = row - 1; 0 <= i ; i--){
                if(ej == -1 && wallpaper[i].charAt(j) == '#'){
                    ej = j + 1;
                    break;
                }
            }
        }

        int[] answer = {si, sj, ei, ej};
        return answer;
    }
}